datafusion/expr/src/function.rs

```diff
@@ -67,6 +88,24 @@ impl<'a> AccumulatorArgs<'a> {
             sort_exprs,
         }
     }
+
+    /// Return a not yet implemented error if IGNORE NULLs is true
+    pub fn check_ignore_nulls(&self, name: &str) -> Result<()> {
+        if self.ignore_nulls {
```

Comment:
Should we check `!self.ignore_nulls`?

Reply:
I think it is confusing for the user to understand whether they need to check or not.
I think rename it to `disable_xxx` help or change the logic an rename it to `enable_xxx`

Reply:
> Should we check `!self.ignore_nulls`?

I don't think we can do this, because the ignore_nulls is true for the following queries

```sql
SELECT avg(x) FROM ...;
SELECT avg(x) RESPECT NULLS FROM ...;
```

In other words, it is the default even when the user doesn't explicitly specify the handling

Reply:
> I think it is confusing for the user to understand whether they need to check or not. I think rename it to `disable_xxx` help or change the logic an rename it to `enable_xxx`

I don't quite understand this suggestion -- are you suggesting rename `check_ignore_nulls` to `disable_ignore_nulls`?

Reply:
> There is also a problem because if they forget to add `check_ignore_nulls/check_order_by` in the accumulator, they can still run the function successfully. This approach does not _force_ the user to check their options because datafusion implements them, not the user.

That is a good point --- in fact it actually affects built in aggregates too today

```sql
❯ select count(*) from (values (1), (null), (2));
+----------+
| COUNT(*) |
+----------+
| 3        |
+----------+
1 row in set. Query took 0.039 seconds.

❯ select count(*) IGNORE NULLS from (values (1), (null), (2));
+----------+
| COUNT(*) |
+----------+
| 3        |
+----------+
1 row in set. Query took 0.001 seconds.
```

I think this is a sepate issue, and not made worse by this PR -- I filed https://github.com/apache/arrow-datafusion/issues/9924 to track. I suggest we work on improving it as a follow on PR

Reply:
> > I think it is confusing for the user to understand whether they need to check or not. I think rename it to `disable_xxx` help or change the logic an rename it to `enable_xxx`
> 
> I don't quite understand this suggestion -- are you suggesting rename `check_ignore_nulls` to `disable_ignore_nulls`?

yes, that is what I suggest.

Reply:
> I think this is a sepate issue, and not made worse by this PR -- I filed #9924 to track. I suggest we work on improving it as a follow on PR

But I think if we implement these for UDFImpl, 
```rust
fn support_ignore_nulls() -> bool
fn support_ordering() -> bool
```

we probably don't need the `check_ignore_nulls`, because we can check it for them!